draft-kwbdgrr-tsvwg-net-collab-rqmts.md

```diff
@@ -423,7 +423,11 @@ It is important that not every flow be prioritized; otherwise, the network devol
 
 Such a mechanism might be simple, for example, a cellular network might allow one flow from a subscriber to declare itself as important; other flows with that subscriber are denied attempts to prioritize themselves. However, the network cannot identify whether the prioritized flow is legitimate or malicious.
 
-Requirement: REQ-??: *need text here*
+Requirements:
+
+  REQ-SIGNAL-VALIDATION: The network/OS needs to ensure that the user/client signaling of priority is unique for every type of traffic within the same flow, thereby avoiding prioritizing of all the streams/traffic the same way.
+
+  REQ-CLIENT-VALIDATION: The network needs to ensure the signal is coming from the same user/client that is part of the 5-tuple flow. This is to ensure no other application influences the priority of another application's flow.
```

Comment:
I purposely didn't include any signaling or host involvement in the process so as not to get into implementation details.

Reply:
line 428 / requirement is not clear how it would work. Assuming that the e2e flow is encrypted, what would the user/client be able to signal about 'type of traffic within a flow' (i.e., to the network every packet looks the same). For example, in QUIC, only the connection id and spin bit are not encrypted and neither has meaning for packet priority. It would seem that only the server can indicate if a packet (or frame) has higher priority.

Reply:
The E2E flow is encrypted (DTLS) but the signaling of metadata would lie outside of the DTLS payload... There should be a mechanism through which the network should be able to decipher the metadata (getting into the mechanism will be moving into implementation stuff and hence, I didn't get into it here). The metadata will obfuscated/encrypted by the server and the key will be sent to the client over the encrypted channel. The client would then share the key with the network/OS (in band or out of band). This is another use of client-to-network signaling that we had in the document but removed since it goes into implementation details. So left it as a requirement. Implementation would be ideally part of a different ID. Metadata can be filled into the CID for QUIC(https://datatracker.ietf.org/doc/draft-wing-cidfi/) or UDP options or anything else...

There would be a <metadata,priority> pair for every flow and not all the metadata should be allowed to have highest priority is what being conveyed here, although how to design that is again getting into implementation part and so I strayed away from it too.

Reply:
Thanks, I follow now, and agree with the proposal